docs/proposals/policy.md

```diff
@@ -0,0 +1,251 @@
+
+The other problem you have is modification.  If you suddenly realize that it is too expensive to have your "vets" able to watch all of your "cows" and "sheep", you may want to change the rules to allow only "gets".  Because you instantiated PolicyRules for each Team/Namespace combination, you need to find them and update them all.  There's still a need to have the capability for an absolute rule, but it wasn't appropriate to explode it out in this case.  That gets you an object that looks something like RoleBinding where you take your set of PolicyRules and bind them a particular team/namespace while still maintain a reference to a set of rules, not a copy of them.
+
+#### Why allow TeamNames instead of simply saying that RoleBindings only have a list of UserNames?  
+Ease of maintenance and separation of powers.  Teams span namespaces, but RoleBindings do not.  If you specify the full set of users on every RoleBinding, then you have to repeat it for every namespace.  If that membership changes, you now have to update it in multiple locations.  As for the separation of powers, if you have rights to modify a RoleBinding, you have the power to change its permissions (RoleTemplateName) instead of only having the power to change its membership.  It seems reasonable to allow someone to modify team membership ("chubba is mechanic") without also having the ability to modify permissions associated with the team itself ("mechanics are super-admins").  Having Teams distinct from Roles makes it easy to express this in policy.
+
+#### Why do I have to specify a RoleBinding in each namespace it applies to?
+Because etcd doesn't have a query engine or referential integrity.  For a given request, we need to quickly locate all policy rules that apply to it.  We have structured our etcd keys to be quickly listable based on namespace.  By storing all rules that apply to a particular request under two key paths (namespaced rules or global rules), that retrieval is relatively efficient.  Storing them any other way would require an efficient index.
+
+### One off endpoints
+#### allowedentities?namespace=<namespace>
+Given a namespace which users, teams, and accounts can see some resource contained in that namespace.  This API is not guaranteed to accurate results.  It is possible that a user returned in the list, could be denied access to any resource in the namespace.  This can happen when a user is granted allow rights for something the namespace, but a deny rule later stomps on it.  Barring weird cases like that, it will work.
+
+#### cando?request=<request>
+Can the authenticated user perform a given request?  Returns true or false.  This API is guaranteed to be accurate.
```

Comment:
What is true or false in HTTP status codes?  Do I get back a HTTP 200 with a document that says "true/false", is there a JSON response structure so we need a struct definition defined somewhere?  Is it just 200 ok, and 409 conflict if not?